### PR TITLE
Added Caching

### PR DESCRIPTION
**Pull request by MrRobot9**
Caching the responses , which will take only 1-2 seconds instead of 10, and will run instantly on next launch. The responses with headers and cookies will be transparently cached to demo_cache.sqlite database.

### CODE REVIEW

**Comment by swapagarwal** [comment]
@MrRobot9 The module looks perfect!
Also, requests are made inside `modules` folder also. I suppose the cache will have to be installed there also. Common cache for all modules or independent?
However, not all responses can be cached, e.g. currency. Basically, we can only cache data if it's not real-time and where caching is not against the policy of using the API.

**Comment by MrRobot9** [comment]
It works for all modules, whichever modules we don't want to cache we can add a  `with requests_cache.disabled():`  above the request , can you list which and all needs to be cached, and which and all doesnt require caching ?

**Comment by swapagarwal** [comment]
@MrRobot9 I'd prefer a precautionary approach, i.e. adding `with requests_cache.enabled():` above requests that can be cached.
Can you look into the modules, and prepare an initial draft of requests that can be cached? Then, we can review it together. 😄

**Comment by MrRobot9** [comment]
@swapagarwal Yaa, guess it's there, did it work for xkcd module?? These requests can be cached
1)anime
2)book
3)dictionary
4)video
5)music
6)movie

**Comment by swapagarwal** [comment]
@MrRobot9 Yes, it's currently working for `xkcd`. Thanks man!

I propose we keep two types of caches: hourly and daily.

#### Hourly cache (data may change):
music
video
xkcd

#### Daily cache (data is very less likely to change):
anime
book
dictionary
movie
wiki

Your views?

**Comment by MrRobot9** [comment]
@swapagarwal  That'd be good

**Comment by swapagarwal** [comment]
Cool, you can take [xkcd ](https://github.com/swapagarwal/JARVIS-on-Messenger/blob/master/modules/src/xkcd.py#L10) module as reference, changing `expire_after` accordingly.

**Comment by MrRobot9** [comment]
@swapagarwal  Yeah sure !! I will update the pull request.

**Comment by MrRobot9** [comment]
@swapagarwal I have pushed the new changes.

**Comment by swapagarwal** [comment]
Hey @MrRobot9, can you remove caching in `currency` module? 😛 
Also, remove changes to files other than `modules/*`. So you should see only 8 changed files here: https://github.com/swapagarwal/JARVIS-on-Messenger/pull/136/files, currently there are 11!

File: .gitignore [hunk 1]
```diff
@@ -60,3 +60,6 @@ target/
 
 #Ipython Notebook
 .ipynb_checkpoints
+
+#Cache
+*.sqlite
```
Comment (swapagarwal) [inline .gitignore]: Remove this for now.

Comment (swapagarwal) [inline .gitignore]: Remove these changes for now.

File: modules/src/xkcd.py [hunk 2]
```diff
@@ -7,7 +7,7 @@
 def process(input, entities=None):
     output = {}
     try:
-        with requests_cache.enabled('xkcd_cache', backend='sqlite', expire_after=3600):
+        with requests_cache.enabled('hourly_cache', backend='sqlite', expire_after=3600):
```
Comment (swapagarwal) [inline modules/src/xkcd.py]: Let's use module names for the caches, so no need to change this file!

File: modules/src/anime.py [hunk 3]
```diff
@@ -1,14 +1,16 @@
...
 
 def process(input, entities):
     output = {}
     try:
-        anime = entities['anime'][0]['value']
-        r = requests.get('https://hummingbird.me/api/v1/search/anime', params={
-            'query': anime
-        })
-        data = r.json()
+        with requests_cache.enabled('daily_cache', backend='sqlite', expire_after=86400):
```
Comment (swapagarwal) [inline modules/src/anime.py]: rename to anime_cache

File: modules/src/anime.py [hunk 4]
```diff
@@ -1,14 +1,16 @@
...
+            anime = entities['anime'][0]['value']
```
Comment (swapagarwal) [inline modules/src/anime.py]: keep this line before caching as this is independent

File: modules/src/currency.py [hunk 5]
```diff
@@ -1,15 +1,17 @@
 import requests
+import requests_cache
+
```
Comment (swapagarwal) [inline modules/src/currency.py]: remove extra line

File: modules/src/currency.py [hunk 6]
```diff
@@ -1,15 +1,17 @@
...
-        to_currency = entities['to_currency'][0]['value'].upper()
-        r = requests.get('http://api.fixer.io/latest?base=' + from_currency)
-        data = r.json()
+        with requests_cache.enabled('daily_cache', backend='sqlite', expire_after=86400):
+            from_currency = entities['from_currency'][0]['value'].upper()
+            to_currency = entities['to_currency'][0]['value'].upper()
+            r = requests.get('http://api.fixer.io/latest?base=' + from_currency)
+            data = r.json()
         conversion_rate = data['rates'][to_currency]
-
```
Comment (swapagarwal) [inline modules/src/currency.py]: keep the separation line

File: modules/src/music.py [hunk 7]
```diff
@@ -1,14 +1,17 @@
...
     output = {}
     try:
-        music = entities['music'][0]['value']
-        r = requests.get('https://api.spotify.com/v1/search?q=' + music + '&type=track')
-        data = r.json()
+        with requests_cache.enabled('hourly_cache', backend='sqlite', expire_after=3600):
+            music = entities['music'][0]['value']
+            r = requests.get('https://api.spotify.com/v1/search?q=' + music + '&type=track')
+            data = r.json()
+            print r.from_cache
```
Comment (swapagarwal) [inline modules/src/music.py]: ??

File: modules/src/wiki.py [hunk 8]
```diff
@@ -1,13 +1,15 @@
...
 from templates.text import TextTemplate
 
 def process(input, entities):
     output = {}
     try:
-        query = entities['wiki'][0]['value']
-        data = wikipedia.page(query)
+        with requests_cache.enabled('daily_cache', backend='sqlite', expire_after=86400):
+            query = entities['wiki'][0]['value']
+            data = wikipedia.page(query)
```
Comment (swapagarwal) [inline modules/src/wiki.py]: `requests` library is not even used here, so no use of `requests_cache` here!

File: modules/src/music.py (hunk 7 above)
Comment (MrRobot9) [inline modules/src/music.py]: Anything wrong ??

Comment (swapagarwal) [inline modules/src/music.py]: I think the print statement above was for debugging purposes.

Comment (MrRobot9) [inline modules/src/music.py]: @swapagarwal Yeah😅😅 I  removed it, is everything fine now?

**Comment by swapagarwal** [comment]
Thanks @MrRobot9! 👍

**Pull request merged**